WalletWasabi.Fluent/ViewModels/Wallets/CoinJoinStateViewModel.cs

```diff
@@ -258,6 +269,58 @@ private void ConfigureStateMachine()
 			.OnExit(() => LeftText = "");
 	}
 
+	private void RefreshButtonAndTextInStateStoppedOrPaused()
+	{
+		// Set visibility of Play button.
+		if (IsAutoCoinJoinEnabled)
+		{
+			PlayVisible = true;
+		}
+		else
+		{
+			if (AreAllCoinsPrivate)
+			{
+				PlayVisible = false;
+			}
+			else
+			{
+				PlayVisible = true;
+			}
+		}
+
+		// Set the message,
+		if (IsAutoCoinJoinEnabled)
+		{
+			CurrentStatus = PauseMessage;
+		}
+		else
+		{
+			if (!AreAllCoinsPrivate)
+			{
+				CurrentStatus = StoppedMessage;
+			}
+
+			// No compulsory update, we do not touch the message. If it was AllPrivateMessage we will keep that.
+		}
+
+		// Set the LeftText.
+		if (IsAutoCoinJoinEnabled)
+		{
+			LeftText = CoinJoinStateViewModel.PressPlayToStartMessage;
+		}
+		else
+		{
+			if (!AreAllCoinsPrivate)
+			{
+				LeftText = CoinJoinStateViewModel.PressPlayToStartMessage;
+			}
+			else
+			{
+				LeftText = "";
+			}
+		}
```

Comment:
Method can be simplified to avoid repeated if-else statements, improve readability, and fix CodeScene.

```suggestion
		if (IsAutoCoinJoinEnabled)
		{
			PlayVisible = true;
			CurrentStatus = PauseMessage;
			LeftText = PressPlayToStartMessage;
		}
		else if (AreAllCoinsPrivate)
		{
			PlayVisible = false;
			LeftText = "";
			// No compulsory update, we do not touch the message. If it was AllPrivateMessage we will keep that.
		}
		else
		{
			PlayVisible = true;
			CurrentStatus = StoppedMessage;
			LeftText = PressPlayToStartMessage;
		}
```

Reply:
I was so into the business logic... Gj